install/infra/terraform/gke/variables.tf

```diff
@@ -0,0 +1,81 @@
+variable "project" {
+  type        = string
+  description = "The project ID to create the cluster."
+}
+
+variable "kubeconfig" {
+  type        = string
+  description = "Path to write the kubeconfig output to"
+  default     = "./kubeconfig"
+}
+
+variable "region" {
+  type        = string
+  description = "The region to create the cluster."
+  default     = "europe-west1"
+}
+
+variable "zone" {
+  type        = string
+  description = "The zone to create the cluster."
+  default     = "b"
+}
+
+variable "kubernetes_version" {
+  type        = string
+  description = "Kubernetes version to be setup"
+  default     = "1.21.11-gke.1100"
+}
+
+variable "name" {
+  type        = string
+  description = "The name of the cluster."
+  default     = "sh-test"
+}
+
+variable "workspaces_machine_type" {
+  type        = string
+  description = "Type of the node compute engines for workspace nodepool."
+  default     = "n2-standard-8"
+}
+
+variable "services_machine_type" {
+  type        = string
+  description = "Type of the node compute engines for services nodepool."
+  default     = "n2-standard-4"
+}
+
+variable "min_count" {
+  type        = number
+  description = "Minimum number of nodes in the NodePool. Must be >=0 and <= max_node_count."
+  default     = 1
+}
+
+variable "max_count" {
+  type        = number
+  description = "Maximum number of nodes in the NodePool. Must be >= min_node_count."
+  default     = 50
```

Comment:
Small comment: a default max_count of 50 for Gitpod services seems pretty high, especially for tests. If we have a reasonable estimate of the number of nodes spun up during an average test run then we may be able to constrain this lower.

In addition, it may be worthwhile to have separate max_count values for the services and workspaces node pools.

Reply:
Out currently suggested `max_count` value in the guides are `100`, I wanted to bring it a bit down if we are using default but keeping it configurable. We can configure this value in the [`main.tf` used for tests](https://github.com/gitpod-io/gitpod/pull/10365/files/5832150fbf31b334b333493f86a2ad5b7ef0dcd1#diff-a39dd45b31fd7a552e7b9786c9bcc9939453bbbbd20dc31835760bc78a56f4eb). I believe it is safe to assume that `install/infra/terraform/` will have more production ready code and `install/tests` has test code.